File: 해시/42577/42577.py
def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book)):
        for j in range(i + 1, len(phone_book)):
            if phone_book[i][0] != phone_book[j][0]:
                break
            if len(phone_book[i]) >= len(phone_book[j]):
                break
            if phone_book[i] == phone_book[j][:len(phone_book[i])]:
                return False            
    return True

File: 해시/42577/42577.cpp
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

bool solution(vector<string> phone_book)
{
   sort(phone_book.begin(), phone_book.end());
   for (int i = 0; i < phone_book.size(); ++i)
      for (int j = i + 1; j < phone_book.size(); ++j)
         if (phone_book[i].size() >= phone_book[j].size() or phone_book[i][0] != phone_book[j][0])
            break;
         else if (phone_book[i] == string(phone_book[j].begin(), phone_book[j].begin() + phone_book[i].size()))
            return false;
   return true;
}